src/Sentry/Internal/Extensions/JsonExtensions.cs

```diff
@@ -156,7 +156,21 @@ public static void Deconstruct(this JsonProperty jsonProperty, out string name,
 
         foreach (var (name, value) in json.EnumerateObject())
         {
+
+            switch (value.ValueKind)
+            {
+                case JsonValueKind.Number:
+                    value.TryGetDouble(out double val);
+                    result[name] = val.ToString();
+                    break;
+                case JsonValueKind.True:
+                case JsonValueKind.False:
+                    result[name] = value.ValueKind.ToString();
+                    break;
+                default:
             result[name] = value.GetString();
+                    break;
+            }
```

Comment:
I changed these because, while not directly linked to the issue at hand: as I was calling `SentrySdk.CauseCrash(CrashType.JavaBackgroundThread);` many times, I noticed that a `The requested operation requires an element of type 'String', but the target element has type 'Number'.' when doing jsonElement.GetString` many times. 

I strongly feel like this belongs in this PR because this fix could provoke it more frequently

Reply:
@bricefriha where does the reference to `GetStringDictionaryOrNull` come from when you're calling` `SentrySdk.CauseCrash`?

It seems odd to to an implicit conversion from into to string in `GetStringDictionaryOrNull` (but only for that type, to make a very specific scenario work). I'm wondering if it's more appropriate to replace that call with `GetDictionaryOrNull` and then make an explicit conversion in the calling code.